dnfdaemon-server/services/base/base.cpp

```diff
@@ -35,14 +35,14 @@ void Base::dbus_register() {
     auto dbus_object = session.get_dbus_object();
     dbus_object->registerMethod(
         dnfdaemon::INTERFACE_BASE, "read_all_repos", "", "b", [this](sdbus::MethodCall call) -> void {
```

Comment:
Only having a brief look right now but the `call` argument of the lambda should be passed by reference, shouldn't it?

Reply:
According to https://github.com/Kistler-Group/sdbus-cpp/blob/e16ffb1288e591526efe25e00c0047c9e21fb5ed/include/sdbus-c%2B%2B/IObject.h#L76 and https://github.com/Kistler-Group/sdbus-cpp/blob/e16ffb1288e591526efe25e00c0047c9e21fb5ed/include/sdbus-c%2B%2B/TypeTraits.h#L56 the `call` argument should be passed by value.

Reply:
I wonder if that's actually a mistake? You can see all the other callbacks have their argument passed as reference and it really makes no sense to me to pass by value? Though maybe there's some special reason that I just don't see... Though it's probably not much of a problem.

FWIW I also don't see why the code would segfault with the moves, did you find out the exact reason?

Reply:
Sorry, I'm probably missing something. But all the callbacks in `registerMethod()` calls looks the same.

What fixed the segfaults (and valgrind errors) was the change in `threads_manager.hpp:handle_signal / handle_method`. Specifically passing `service` to the worker thread explicitly by reference using `std::ref(service)`. Without this `service` was copied when the thread started and that caused the problems.

Reply:
> Sorry, I'm probably missing something. But all the callbacks in registerMethod() calls looks the same.

I meant the other callbacks at https://github.com/Kistler-Group/sdbus-cpp/blob/e16ffb1288e591526efe25e00c0047c9e21fb5ed/include/sdbus-c%2B%2B/TypeTraits.h#L56 through to line 60. Seems like a possible omission to use a reference in this callback too.

But this looks good.